moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp

```diff
@@ -35,167 +35,173 @@
 /* Author: Ioan Sucan, Dave Coleman, Robert Haschke */
 
 #include "moveit_fake_controllers.h"
-#include <ros/ros.h>
 #include <moveit/robot_state/robot_state.h>
 #include <moveit/robot_model_loader/robot_model_loader.h>
-#include <sensor_msgs/JointState.h>
+#include <sensor_msgs/msg/joint_state.hpp>
 #include <pluginlib/class_list_macros.hpp>
-#include <ros/console.h>
 #include <map>
 #include <iterator>
 
 namespace moveit_fake_controller_manager
 {
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit.plugins.moveit_fake_controller_manager");
 static const std::string DEFAULT_TYPE = "interpolate";
 static const std::string ROBOT_DESCRIPTION = "robot_description";
 
 class MoveItFakeControllerManager : public moveit_controller_manager::MoveItControllerManager
 {
 public:
-  MoveItFakeControllerManager() : node_handle_("~")
+  MoveItFakeControllerManager() : MoveItControllerManager()
   {
-    if (!node_handle_.hasParam("controller_list"))
+  }
+
+  void initialize(const rclcpp::Node::SharedPtr& node)
+  {
+    // TODO(henningkayser): use flexible base
+    const std::string PARAM_BASE_NAME = "moveit_fake_controller_manager";
+    node_ = node;
+    if (!node_->has_parameter(PARAM_BASE_NAME + ".controller_names"))
     {
-      ROS_ERROR_STREAM_NAMED("MoveItFakeControllerManager", "No controller_list specified.");
+      RCLCPP_ERROR(LOGGER, "No controller_names specified.");
       return;
     }
 
-    XmlRpc::XmlRpcValue controller_list;
-    node_handle_.getParam("controller_list", controller_list);
-    if (controller_list.getType() != XmlRpc::XmlRpcValue::TypeArray)
+    rclcpp::Parameter controller_names_param;
+    node_->get_parameter(PARAM_BASE_NAME + ".controller_names", controller_names_param);
+    if (controller_names_param.get_type() != rclcpp::ParameterType::PARAMETER_STRING_ARRAY)
     {
-      ROS_ERROR_NAMED("MoveItFakeControllerManager", "controller_list should be specified as an array");
+      RCLCPP_ERROR(LOGGER, "Parameter controller_names should be specified as a string array");
       return;
     }
 
     /* by setting latch to true we preserve the initial joint state while other nodes launch */
-    bool latch = true;
-    pub_ = node_handle_.advertise<sensor_msgs::JointState>("fake_controller_joint_states", 100, latch);
+    pub_ = node_->create_publisher<sensor_msgs::msg::JointState>("fake_controller_joint_states", 100);
 
     /* publish initial pose */
-    XmlRpc::XmlRpcValue initial;
-    if (node_handle_.getParam("initial", initial))
-    {
-      sensor_msgs::JointState js = loadInitialJointValues(initial);
-      js.header.stamp = ros::Time::now();
-      pub_.publish(js);
-    }
+    // TODO: codebase wide refactoring for XmlRpc
```

Comment:
I think we should add a basic implementation for this. Loading this parameter without XmlRpc shouldn't be too much an issue, no?

Reply:
port e9b3c1d